// nativescript-plugin-firebase/crashlytics/crashlytics.ios.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var crashlytics_common_1 = require("./crashlytics-common");
function sendCrashLog(exception) {
    if (isCrashlyticsAvailable()) {
        FIRCrashlytics.crashlytics().recordError(exception);
    }
}
exports.sendCrashLog = sendCrashLog;
function log(msg, tag, priority) {
    if (isCrashlyticsAvailable()) {
        if (tag) {
            FIRCrashlytics.crashlytics().log(tag + " - " + msg);
        }
        else {
            FIRCrashlytics.crashlytics().log(msg);
        }
    }
}
exports.log = log;
function setString(key, value) {
    if (isCrashlyticsAvailable()) {
        FIRCrashlytics.crashlytics().setCustomValueForKey(value, key);
    }
}
exports.setString = setString;
function setBool(key, value) {
    if (isCrashlyticsAvailable()) {
        FIRCrashlytics.crashlytics().setCustomValueForKey(value, key);
    }
}
exports.setBool = setBool;
function setFloat(key, value) {
    if (isCrashlyticsAvailable()) {
        FIRCrashlytics.crashlytics().setCustomValueForKey(value, key);
    }
}
exports.setFloat = setFloat;
function setInt(key, value) {
    if (isCrashlyticsAvailable()) {
        FIRCrashlytics.crashlytics().setCustomValueForKey(value, key);
    }
}
exports.setInt = setInt;
function setDouble(key, value) {
    if (isCrashlyticsAvailable()) {
        FIRCrashlytics.crashlytics().setCustomValueForKey(value, key);
    }
}
exports.setDouble = setDouble;
function setUserId(id) {
    if (isCrashlyticsAvailable()) {
        FIRCrashlytics.crashlytics().setUserID(id);
    }
}
exports.setUserId = setUserId;
function crash() {
    if (isCrashlyticsAvailable()) {
        NSException.exceptionWithNameReasonUserInfo("FIRCrashlytics", "test crash", null).raise();
    }
}
exports.crash = crash;
function setCrashlyticsCollectionEnabled(enabled) {
    if (isCrashlyticsAvailable()) {
        FIRCrashlytics.crashlytics().setCrashlyticsCollectionEnabled(enabled);
    }
}
exports.setCrashlyticsCollectionEnabled = setCrashlyticsCollectionEnabled;
function isCrashlyticsAvailable() {
    if (typeof (FIRCrashlytics) === "undefined") {
        console.log(crashlytics_common_1.ENABLE_CRASHLYTICS_HINT);
        return false;
    }
    return true;
}
