// nativescript-plugin-firebase/crashlytics/crashlytics.android.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var crashlytics_common_1 = require("./crashlytics-common");
function sendCrashLog(exception) {
    if (isCrashlyticsAvailable()) {
        com.google.firebase.crashlytics.FirebaseCrashlytics.getInstance().recordException(exception);
    }
}
exports.sendCrashLog = sendCrashLog;
function log(msg, tag, priority) {
    if (isCrashlyticsAvailable()) {
        if (tag && priority) {
            var fullMessage = priority + '/' + tag + ': ' + msg;
            com.google.firebase.crashlytics.FirebaseCrashlytics.getInstance().log(fullMessage);
        }
        else {
            com.google.firebase.crashlytics.FirebaseCrashlytics.getInstance().log(msg);
        }
    }
}
exports.log = log;
function setString(key, value) {
    if (isCrashlyticsAvailable()) {
        com.google.firebase.crashlytics.FirebaseCrashlytics.getInstance().setCustomKey(key, value);
    }
}
exports.setString = setString;
function setBool(key, value) {
    if (isCrashlyticsAvailable()) {
        com.google.firebase.crashlytics.FirebaseCrashlytics.getInstance().setCustomKey(key, value);
    }
}
exports.setBool = setBool;
function setFloat(key, value) {
    if (isCrashlyticsAvailable()) {
        com.google.firebase.crashlytics.FirebaseCrashlytics.getInstance().setCustomKey(key, value);
    }
}
exports.setFloat = setFloat;
function setInt(key, value) {
    if (isCrashlyticsAvailable()) {
        com.google.firebase.crashlytics.FirebaseCrashlytics.getInstance().setCustomKey(key, value);
    }
}
exports.setInt = setInt;
function setDouble(key, value) {
    if (isCrashlyticsAvailable()) {
        com.google.firebase.crashlytics.FirebaseCrashlytics.getInstance().setCustomKey(key, value);
    }
}
exports.setDouble = setDouble;
function setUserId(id) {
    if (isCrashlyticsAvailable()) {
        com.google.firebase.crashlytics.FirebaseCrashlytics.getInstance().setUserId(id);
    }
}
exports.setUserId = setUserId;
function crash() {
    if (isCrashlyticsAvailable()) {
        throw new java.lang.RuntimeException('Test Crash');
    }
}
exports.crash = crash;
function setCrashlyticsCollectionEnabled(enabled) {
    com.google.firebase.crashlytics.FirebaseCrashlytics.getInstance().setCrashlyticsCollectionEnabled(new java.lang.Boolean(enabled));
}
exports.setCrashlyticsCollectionEnabled = setCrashlyticsCollectionEnabled;
function isCrashlyticsAvailable() {
    if (typeof (com.google.firebase.crashlytics.FirebaseCrashlytics) === "undefined" || typeof (com.google.firebase.crashlytics.FirebaseCrashlytics.getInstance()) === "undefined") {
        console.log(crashlytics_common_1.ENABLE_CRASHLYTICS_HINT);
        return false;
    }
    return true;
}
